compiler/z/codegen/OMRPeephole.cpp

```diff
@@ -544,6 +549,40 @@ OMR::Z::Peephole::attemptToReduceLToICM(TR::Instruction* cursor)
    return performed;
    }
 
+bool
+OMR::Z::Peephole::attemptToReduceLLCToLLGC(TR::Instruction* cursor)
+   {
+   TR::Instruction *current = cursor->getNext();
+   auto mnemonic = current->getOpCodeValue();
+
+   if (mnemonic == TR::InstOpCode::LGFR || mnemonic == TR::InstOpCode::LLGTR)
+      {
+      TR::Register *llcTgtReg = ((TR::S390RRInstruction *) cursor)->getRegisterOperand(1);
+
+      TR::Register *curSrcReg = ((TR::S390RRInstruction *) current)->getRegisterOperand(2);
+      TR::Register *curTgtReg = ((TR::S390RRInstruction *) current)->getRegisterOperand(1);
```

Comment:
Just confirming this is correct as both `curTgtReg` and `llcTgtReg` are operand 1.

Reply:
Actually, if that is true then the condition below doesn't need the `llcTgtReg == curTgtReg` check as it will always be true.

Reply:
Yes those are correct, however the values are not the same. One extracts it from `current` one extracts from `cursor`, and `current = cursor->getNext()`. Poor variable names IMO. I just copied it from existing code and did not make any changes. I've changed the variable name of `current` to `nextInst` so it is more clear. Fixed in 92617ef

Reply:
Oh, duh.  Yeah, I missed that.  Sorry.